modules/test_user_dashboard/app/controllers/test_user_dashboard/oauth_controller.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require_dependency 'test_user_dashboard/application_controller'
+
+module TestUserDashboard
+  class OAuthController < ApplicationController
+    include Warden::GitHub::SSO
+
+    before_action :authenticate!, only: [:index]
+    before_action :authorize!, only: [:index]
+
+    # Action for the redirect_uri for GitHub OAuth
+    def index
+      redirect_to "#{url}/signin"
+    end
+
+    # Action to authorize TUD requests
+    def authenticated_and_authorized?
+      render json: @current_user if authorized?
+    end
+
+    def logout
+      if authenticated?
+        Rails.logger.info("TUD user logged out: #{github_user_details}")
+        warden.logout(:tud)
+      end
+
+      redirect_to url
+    end
+
+    private
+
+    def url
+      return 'https://tud.vfs.va.gov' if Settings.vsp_environment == 'staging'
```

Comment:
Yeah, the intent is to make things easier when running locally. I can't use `Settings.test_user_dashboard.env == 'staging'` because that will set the URL to the TUD app so I had to use `Settings.vsp_environment`.

I took inspiration from the TUD app that sets the URL based on environment to make it easier to run locally.